src/ledger/mod.rs

```diff
@@ -105,6 +106,54 @@ impl Ledger {
         Ok(())
     }
 
+    #[instrument(name = "ledger.utxo_dropped", skip(self, tx))]
+    pub async fn utxo_dropped(
+        &self,
+        tx: Transaction<'_, Postgres>,
+        tx_id: LedgerTransactionId,
+        detected_txn_id: LedgerTransactionId,
+        ledger_account_ids: WalletLedgerAccountIds,
```

Comment:
This should be derived from the entries of the `detected_txn_id` and not passed in.